### PR TITLE
runtime: migrate stake program to core bpf

### PR DESCRIPTION
**Pull request by 2501babe**
#### Problem
with the bpf stake program preparing for audits, we would like to add the runtime hooks to deploy it when ready

#### Summary of Changes
add support for the core bpf stake program migration

Feature Gate Issue: #3654

### CODE REVIEW

**Review by buffalojoec — DISMISSED** [review]
Lgtm 🚢

File: sdk/feature-set/src/lib.rs [hunk 1]
```diff
@@ -1104,6 +1108,7 @@ lazy_static! {
         (disable_account_loader_special_case::id(), "Disable account loader special case #3513"),
         (accounts_lt_hash::id(), "enables lattice-based accounts hash #3333"),
         (enable_secp256r1_precompile::id(), "Enable secp256r1 precompile SIMD-0075"),
+        (migrate_stake_program_to_core_bpf::id(), "Migrate Stake program to Core BPF #XXXX"),
```
Comment (joncinque) [inline sdk/feature-set/src/lib.rs]: nit:
```suggestion
        (migrate_stake_program_to_core_bpf::id(), "Migrate Stake program to Core BPF SIMD-0196 #3655"),
```

**Review by joncinque — COMMENTED** [review]
Looks good to me! Just the comment about the doc for the feature

**Review by joncinque — APPROVED** [review]
Great, thanks!

**Review by buffalojoec — APPROVED** [review]

**Pull request merged**